app/App.tsx

```diff
@@ -78,8 +79,12 @@ const App: FC = () => {
         // solves an issue of this component's local state fighting with the redux store.
         const { selectedAccount } = store.getState();
         if (selectedAccount !== initialReduxStoreState.selectedAccount) {
-          await fetchBalance(selectedAccount);
-          await fetchLogs(selectedAccount);
+          // check wallet status to see if we need to update account
+          const isSynced = await isAccountSynced({ accountId: selectedAccount.account.accountId });
+          if (!isSynced) {
+            await fetchBalance(selectedAccount);
+            await fetchLogs(selectedAccount);
+          }
```

Comment:
Would `fetchBalance` and `fetchLogs` be better referenced as part of checking `isAccountSynced`? e.g. moving them into that method. It might be better to change the method to be `syncAccount` rather than `isAccountSynced` in that case.

My thinking is that -- if everywhere we call `isAccountSynced`, we're also fetching balance and logs on a `false` value -- it might be better to just wrap that in one method call and use that instead of separating out the checking and fixing functions

Reply:
I think it could be okay either way with a slight preference to leave things as they are. I think it feels a bit easier for someone reading the code to see the control flow of: "check if account synced, if not, then sync", rather than see a function called `syncAccount` and not be aware that there might not be a sync going on